firstandlast.java

```diff
@@ -0,0 +1,44 @@
+O(n),O(1)
+public int search(int[] nums, int target) {
```

Comment:
You are just doing binary search. So time is O(log n).
The logic for firstOccurence() and lastOccurence() is same. I would recommend you to use a flag and avoid code duplication.